### PR TITLE
SO-242 WAF Tweaks For Added Features

### PR DESCRIPTION
**Pull request by StevenPaugh**
Sets ability to write customer curated WAF rules in NAXSI format depending on on-premise customer needs.

Adds ability to disable WAF just in case or based on client needs.

### CODE REVIEW

File: src/_configure_plextrac.sh [hunk 1]
```diff
@@ -207,3 +207,13 @@ function create_volume_directories() {
...
+function configure_waf() {
+  title "Create directory and file for customer curated WAF rules"
+  info "Validating directory and file for customer curated WAF rules"
+  debug "Ensuring directory and file exist for WAF."
+  if test -f "${PLEXTRAC_HOME}/volumes/naxsi-waf/customer_curated.rules"; then
+    debug "WAF Config already exists"
+  else
+    mkdir -vp "${PLEXTRAC_HOME}/volumes/naxsi-waf"
+    echo "## Custom WAF Rules Below" > ${PLEXTRAC_HOME}/volumes/naxsi-waf/customer_curated.rules
+}
```
Comment (Michael-Burke) [inline src/_configure_plextrac.sh:219]: If you want the customer to be able to currate WAF rules, I think we can remove this logic entirely in favor of a manual step(s) we can provide on request. Something along the lines of:

```shell
su - plextrac
mkdir -vp ~/volumes/naxsi-waf
echo "## Custom WAF Rules Below" > ${PLEXTRAC_HOME}/volumes/naxsi-waf/customer_curated.rules
vi docker-compose.override.yml

---
services:
  plextracnginx:
    volumes:
      - naxsi_waf:/etc/nginx/conf.d/customer_curated.rules:r
volumes:
  naxsi_waf:
    driver: local
    driver_opts:
      type: "none"
      o: "bind"
      device: "${PLEXTRAC_HOME:-.}/volumes/naxsi-waf/customer_curated.rules"
---
docker compose -f docker-compose.yml -f docker-compose.override.yml up -d plextracnginx --force-recreate
```

File: static/docker-compose.yml [hunk 2]
```diff
@@ -118,12 +118,14 @@ services:
...
       OVERRIDE_SENTRY_FRONTEND_ENABLED: "${OVERRIDE_SENTRY_FRONTEND_ENABLED-}"
       USE_CUSTOM_CERT: "${USE_CUSTOM_CERT-}"
+      WAF_ENABLED: "${WAF_ENABLED:-true}"
     ports:
     - 0.0.0.0:80:80/tcp
     - 0.0.0.0:443:443/tcp
     restart: always
     volumes:
     - letsencrypt:/etc/letsencrypt:rw
+    - ${PLEXTRAC_HOME:-.}/volumes/naxsi-waf/customer_curated.rules:/etc/nginx/conf.d/customer_curated.rules:r
```
Comment (Michael-Burke) [inline static/docker-compose.yml:128]: Given above, we can remove this unless its expected and required by the container.

Comment (StevenPaugh) [reply static/docker-compose.yml:128]: It is expected by the container. The idea is to populate this file with a comment and at mount the management util combines the pextrac curated rules and any customer curated custom rules into the same file

File: src/_configure_plextrac.sh (hunk 1 above)
Comment (StevenPaugh) [reply src/_configure_plextrac.sh:219]: I'd like to just have this populated with a comment at plextrac configure. I could add logic to look for a customer_curated.rules file in the nginx_init.sh script and if it is there combine the files, and if not then just rename the rules file I include in the image build.  Whatever you think is best is fine by me dude!

**Review by Michael-Burke — DISMISSED** [review]

**Review by Michael-Burke — APPROVED** [review]

**Comment by StevenPaugh** [comment]
Tested this on my dev box. Looks like it should be quite straightforward and work without issue

**Pull request merged**